media/starboard/starboard_renderer.h

```diff
@@ -12,364 +12,138 @@
-             const base::Closure& duration_change_cb,
-             const base::Closure& output_mode_change_cb,
-             const base::Closure& content_size_change_cb) override;
-#endif  // SB_HAS(PLAYER_WITH_URL)
-
-  void Stop(const base::Closure& stop_cb) override;
-  void Seek(TimeDelta time, const SeekCB& seek_cb);
-  bool HasAudio() const override;
-  bool HasVideo() const override;
-
-  float GetPlaybackRate() const override;
-  void SetPlaybackRate(float playback_rate) override;
-  float GetVolume() const override;
-  void SetVolume(float volume) override;
-
-  TimeDelta GetMediaTime() override;
-  ::media::Ranges<TimeDelta> GetBufferedTimeRanges() override;
-  TimeDelta GetMediaDuration() const override;
-#if SB_HAS(PLAYER_WITH_URL)
-  TimeDelta GetMediaStartDate() const override;
-#endif  // SB_HAS(PLAYER_WITH_URL)
-  void GetNaturalVideoSize(gfx::Size* out_size) const override;
-  std::vector<std::string> GetAudioConnectors() const override;
-
-  bool DidLoadingProgress() const override;
-  PipelineStatistics GetStatistics() const override;
-  SetBoundsCB GetSetBoundsCB() override;
-  void SetPreferredOutputModeToDecodeToTexture() override;
+  explicit StarboardRenderer(
+      const scoped_refptr<base::SequencedTaskRunner>& task_runner);
+
+  ~StarboardRenderer() final;
+
+  // Renderer implementation.
+  void Initialize(MediaResource* media_resource,
+                  RendererClient* client,
+                  PipelineStatusCallback init_cb) final;
+  void SetCdm(CdmContext* cdm_context, CdmAttachedCB cdm_attached_cb) final {
+    // TODO(b/328305808): Implement encrypted playback.
+    NOTIMPLEMENTED();
+  }
+  void SetLatencyHint(absl::optional<base::TimeDelta> latency_hint) final {
+    // TODO(b/375271848): Address NOTIMPLEMENTED().
+    NOTIMPLEMENTED();
+  }
+  void SetPreservesPitch(bool preserves_pitch) final {
+    // TODO(b/375271848): Address NOTIMPLEMENTED().
+    NOTIMPLEMENTED();
+  }
+  void SetWasPlayedWithUserActivation(
+      bool was_played_with_user_activation) final {
+    // TODO(b/375271848): Address NOTIMPLEMENTED().
+    NOTIMPLEMENTED();
+  }
+  void Flush(base::OnceClosure flush_cb) final;
+  void StartPlayingFrom(base::TimeDelta time) final;
+  void SetPlaybackRate(double playback_rate) final;
+  void SetVolume(float volume) final;
+  base::TimeDelta GetMediaTime() final;
+  void OnSelectedVideoTracksChanged(
+      const std::vector<DemuxerStream*>& enabled_tracks,
+      base::OnceClosure change_completed_cb) final {
+    // TODO(b/375271848): Address NOTIMPLEMENTED().
+    NOTIMPLEMENTED();
+  }
+  void OnEnabledAudioTracksChanged(
+      const std::vector<DemuxerStream*>& enabled_tracks,
+      base::OnceClosure change_completed_cb) final {
+    // TODO(b/375271848): Address NOTIMPLEMENTED().
+    NOTIMPLEMENTED();
+  }
+  RendererType GetRendererType() final {
+    // Reuse `kRendererImpl` to avoid introducing a new renderer type.
+    return RendererType::kRendererImpl;
```

Comment:
Would we consider to have a new renderer type for Starboard? If yes, maybe we should put a TODO here.

Reply:
Done.